### PR TITLE
Fixing grammer of & improving CA1836 messages.

### PR DESCRIPTION
**Pull request by BrunoJuchli**
Improving the message of PreferIsEmptyOverCountMessage (CA1836):

old  : Prefer 'IsEmpty' over 'Count' to determine whether the object contains or not any items
new: Prefer 'IsEmpty' over 'Count' to determine whether the object is empty

Also slightly improved the german & french translations.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/roslyn-analyzers?pullRequest=4210) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/roslyn-analyzers?pullRequest=4210) before we can accept your contribution.<br/><br/>:x: BrunoJuchli [sign now](https://cla.dotnetfoundation.org/dotnet/roslyn-analyzers?pullRequest=4210)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/roslyn-analyzers?pullRequest=4210) it.</sub>

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.fr.xlf
Comment (Evangelink) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.fr.xlf:1572]: What about "pour déterminer si l'objet est vide" to match the English?

**Review by Evangelink — APPROVED** [review]
Minor suggestion, otherwise LGTM.

Tagging @mavasani

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx
Comment (Evangelink) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx:1414]: @BrunoJuchli Sorry I missed it, the message string should not end with a period.

Comment (mavasani) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx:1414]: Yes, this will cause build failure.

Comment (mavasani) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx:1414]: Also tagging @Jozkee for review.

**Review by mavasani — APPROVED** [review]
LGTM once you remove the period at the end of the new message string and update all the xlf files accordingly.

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.fr.xlf
Comment (BrunoJuchli) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.fr.xlf]: @Evangelink I changed the message as per your suggestion. I also adapted the french version of `MicrosoftNetCoreAnalyzersResources` accordingly.

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx
Comment (BrunoJuchli) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx]: Thank you. I've removed the trailing period in the Message.
I've also done so in the French and in the German translations (even though it did have a trailing period before). I guess that's ok (but the build is not checking translations for correctness in that regard) -- right?
I haven't checked/fixed translations to other languages. I could but then I believe it would probably be best to do this in a separate PR - do you concur?

**Review by mavasani — APPROVED** [review]

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx
Comment (Jozkee) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx:1414]: I think you need to mention two alternatives to properly use "whether".

```suggestion
    <value>Prefer 'IsEmpty' over 'Count' to determine whether the object is empty or not</value>
```

```suggestion
    <value>Prefer 'IsEmpty' over 'Count' to determine whether or not the object is empty</value>
```

We could even rephrase to not use the word
```suggestion
    <value>Prefer 'IsEmpty' over 'Count' to determine if the object is empty</value>
```

Comment (BrunoJuchli) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx:1414]: @Jozkee
I'm not a native speaker so I looked it up.,The first two resources I found don't back that up
( https://english.stackexchange.com/a/48773/398754, http://www.youtube.com/watch?v=Klnroe1UBRs&t=1m23s <-- here making a specific example with whether and stating that "or not"  is not necessary).
So I think it's rather a stylistic choice. Personally I would go for the shorter version unless there's ambiguity.
That would make me lean toward the `if` version (probably also most commonly understood by non native speakers; every programmer will know `if` introduces a condition). Though the guy in the youtube video said that `whether` specifies a binary condition and `if` goes well with multiple options.

Maybe we could have an odd number of native speakers vote on a version? :D

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx
Comment (Evangelink) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx]: @BrunoJuchli You are right, the analyzer only check the English message, title and description entries. I was thinking about checking the other languages but that would probably require to discuss with native speakers to see what they think. Basically we wrote this analyzer to ensure that there is a consistent style in how messages are shown to the user (at least partially).

**Comment by mavasani** [comment]
@BrunoJuchli Can you please rebuild locally to resolve conflicts?

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx
Comment (mavasani) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx:1414]: Lets go with the changes and if required/based on feedback, we can change them in future.

**Comment by BrunoJuchli** [comment]
@mavasani I just rebased on current master and force pushed. Strangely this PR doesn't seem to pick up the change, though stackoverflow states multiple times force-pushing to a branch that has an open PR is getting updated.
Will wait until tomorrow to see if it sync's eventually, otherwise I'll go back to before-rebase version and merge in latest develop.

**Comment by mavasani** [comment]
Closing and re-opening PR to retrigger tests, as we checked in new CI validation step for generated files.

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.cs.xlf
Comment (buyaa-n) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.cs.xlf:1572]: I think the status should be updated to `new` or something so that it can get translated later. Seems this update is done manually instead of generated by build. 

@BrunoJuchli you only need to update `MicrosoftNetCoreAnalyzersResources.resx` and the changes for other languages should be generated automatically with the repo build

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.de.xlf
Comment (buyaa-n) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.de.xlf:1572]: I don't think we should do translations manually

Comment (BrunoJuchli) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.de.xlf]: @buyaa-n ok, since you don't want to have manual translations I've reduced this PR to changing the english text in the resx and then re-generating the xlf files (which re-sets the state to "needs-review-translation").

**Review by mavasani — APPROVED** [review]
LGTM

@buyaa-n Please feel free to review and merge if the changes look good to you. Thanks!

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.de.xlf
Comment (buyaa-n) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.de.xlf]: Thanks @BrunoJuchli, JFYI it is not because i don't like your translation, even if your translation was correct for this language the other languages will be left without translation/update. We normally just add/update the English version and build will update the state so that localization team translate/update them

**Review by buyaa-n — APPROVED** [review]

**Pull request merged**
